### ISSUE TITLE
coap-client fails for PSK 

### ISSUE THREAD

**Issue by boaks** [obgm/libcoap#568]
Somewhere before PR #561, commit 9b39bd9

client.c

```
static ssize_t
cmdline_read_user(char *arg, unsigned char **buf, size_t maxlen) {
  size_t len = strnlen(arg, maxlen);
  if (len) {
    *buf = (unsigned char *)arg;
    /* len is the size or less, so 0 terminate to maxlen */
    buf[len] = '\000';
  }
  /* 0 length Identity is valid */
  return len;
}
```

must be

```
    (*buf)[len] = '\000';
```

otherwise the pointer will be set to zero and PSK fails with

> WARN Identity (PSK) not defined

I open a new issue.

**Comment by mrdeep1** [comment]
Code fixed in #569 (and #561)

**Comment by boaks** [comment]
Great!
